config/web/web.go

```diff
@@ -40,10 +56,31 @@ func (web *Config) ValidateAndSetDefaults() error {
 	} else if web.Port < 0 || web.Port > math.MaxUint16 {
 		return fmt.Errorf("invalid port: value should be between %d and %d", 0, math.MaxUint16)
 	}
+	// Try to load the TLS certificates
+	_, err := web.TLSConfig()
+	if err != nil {
+		return fmt.Errorf("invalid tls config: %s", err)
+	}
 	return nil
 }
 
 // SocketAddress returns the combination of the Address and the Port
 func (web *Config) SocketAddress() string {
 	return fmt.Sprintf("%s:%d", web.Address, web.Port)
 }
+
+// TLSConfig returns a tls.Config object for serving over an encrypted channel
+func (web *Config) TLSConfig() (*tls.Config, error) {
+	if web.tlsConfig == nil && len(web.Tls.CertFile) > 0 && len(web.Tls.KeyFile) > 0 {
+		web.loadTLSConfig()
+	}
+	return web.tlsConfig, web.tlsConfigError
+}
+
+func (web *Config) loadTLSConfig() {
+	cer, err := tls.LoadX509KeyPair(web.Tls.CertFile, web.Tls.KeyFile)
+	if err != nil {
+		web.tlsConfigError = err
+	}
```

Comment:
I'd rather just return the error here instead of persisting the error in the configuration struct

Reply:
Yes, it is a bit awkward. In my first version I directly returned the error. But the issue is that the `TLSConfig()` method is called twice, once in `ValidateAndSetDefaults()` and  once in `controller.go`. So we might try to open the same certificate files twice and create the same error twice. This can only be avoided if we somehow persist the error. Memoization would be a little bit nice because the state would be kept inside the function, but there is no good way to express that in Go, or?

If you see any better way, I'm happy to use it :-)